hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFilePreadReader.java

```diff
@@ -77,6 +78,11 @@ public void run() {
                 block.release();
               }
             }
+            cacheConf.getBlockCache().ifPresent(bc -> {
+              if (bc instanceof BucketCache) {
+                ((BucketCache) bc).fileCacheCompleted(path.getName());
+              }
```

Comment:
IIUC, persistent cache is only for L2 cache. I think, we need to get the instance of L2 cache only if the blockcache is an instance of CombinedBlockCache.